app/services/course/assessment/question/codaveri_problem_generation_service.rb

```diff
@@ -26,28 +26,53 @@ def codaveri_generate_problem
   # Creates a new service codaveri feedback rating object.
   #
   # @param [Course::Discussion::Post::CodaveriFeedback] feedback Feedback to be sent to Codaveri
-  def initialize(assessment, custom_prompt, language, version, difficulty)
+  def initialize(assessment, params, language, version)
+    custom_prompt = params[:custom_prompt]
     @payload = {
       userId: assessment.creator_id.to_s,
       courseName: assessment.course.title,
-      config: {
-        difficulty: difficulty,
-        customPrompt: custom_prompt.length >= 500 ? "#{custom_prompt[0...495]}..." : custom_prompt,
-        testcasesType: 'expression',
-        languageVersion: {
-          language: language,
-          version: version
-        }
+      languageVersion: {
+        language: language,
+        version: version
+      },
+      llmConfig: {
+        customPrompt: (custom_prompt.length >= 500) ? "#{custom_prompt[0...495]}..." : custom_prompt,
+        testcasesType: 'expression'
       },
       requireToken: true,
       tokenConfig: {
         returnResult: true
       }
     }
+
+    return unless params[:is_first_generated_question] == 'false'
+
+    @payload = @payload.merge({
+      problem: {
+        title: params[:title],
+        description: params[:description],
+        templates: [{
+          path: 'main.py',
+          content: params[:template]
+        }],
+        solutions: [{
+          tag: 'solution',
+          files: [{
+            path: 'main.py',
+            content: params[:solution]
+          }]
```

Comment:
Add a TODO here to update the paths to support other languages using the `language` param once it's available

Reply:
done